src/platform/silabs/multi-ota/SiWx917/OTAWiFiFirmwareProcessor.cpp

```diff
@@ -0,0 +1,176 @@
+    {
+        uint32_t mAlignmentNum = (mUnalignmentNum + block.size()) / 16;
+        mAlignmentNum          = mAlignmentNum * 16;
+        mUnalignmentNum        = (mUnalignmentNum + block.size()) % 16;
+        memcpy(&mBlock[requestedOtaMaxBlockSize], &mBlock[mAlignmentNum], mUnalignmentNum);
+        mBlock.reduce_size(mAlignmentNum);
+    }
+    else
+    {
+        mUnalignmentNum = mUnalignmentNum + block.size() - requestedOtaMaxBlockSize;
+        mBlock.reduce_size(requestedOtaMaxBlockSize);
+    }
+
+    OTATlvProcessor::vOtaProcessInternalEncryption(mBlock);
+    block = mBlock;
+#endif //OTA_ENCRYPTION_ENABLE
+
+    if (flag == RPS_HEADER)
+    {
+        memcpy(&writeBuffer, block.data(), kAlignmentBytes);
+        // Send RPS header which is received as first chunk
+        status = sl_si91x_fwup_start(writeBuffer);
+        status = sl_si91x_fwup_load(writeBuffer, kAlignmentBytes);
+        flag   = RPS_DATA;
+        memcpy(&writeDataBuffer, block.data() + kAlignmentBytes, (block.size() - kAlignmentBytes));
+        status = sl_si91x_fwup_load(writeDataBuffer, (block.size() - kAlignmentBytes));
+    }
+    else if (flag == RPS_DATA)
+    {
+        memcpy(&writeDataBuffer, block.data(), block.size());
+        // Send RPS content
+        status = sl_si91x_fwup_load(writeDataBuffer, block.size());
+        if (status != SL_STATUS_OK)
+        {
+            // When TA recived all the blocks it will return SL_STATUS_SI91X_FW_UPDATE_DONE status
+            if (status == SL_STATUS_SI91X_FW_UPDATE_DONE)
+            {
+                mReset = true;
+            }
+            else
+            {
+                ChipLogError(SoftwareUpdate, "ERROR: In HandleProcessBlock sl_si91x_fwup_load() error %ld", status);
+                return CHIP_ERROR_CANCELLED;
+            }
+        }
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OTAWiFiFirmwareProcessor::ProcessDescriptor(ByteSpan & block)
+{
+    ReturnErrorOnFailure(mAccumulator.Accumulate(block));
+    ReturnErrorOnFailure(mCallbackProcessDescriptor(static_cast<void *>(mAccumulator.data())));
+
+    mDescriptorProcessed = true;
+    mAccumulator.Clear();
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OTAWiFiFirmwareProcessor::ApplyAction()
+{
+    // This reboots the device
+    if (mReset)
+    {
+        ChipLogProgress(SoftwareUpdate, "WiFi Device OTA update complete");
+#ifdef SLI_SI91X_MCU_INTERFACE // This is not needed for the 917 SoC; it is required for EFR host applications
+        // send system reset request to reset the MCU and upgrade the m4 image
+        ChipLogProgress(SoftwareUpdate, "SoC Soft Reset initiated!");
+        // Reboots the device
+        sl_si91x_soc_nvic_reset();
+#endif
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR OTAWiFiFirmwareProcessor::FinalizeAction()
+{
+    // TODO: Not requied by 917 OTA updated keeping this function to execute any other command before soft reset of the TA
```

Comment:
Why is it a todo if it is not required?
Why is the padding not necessary for the SiWx917?

Reply:
To-Do is removed from here as this is not required.
 Padding is not required for 917 SoC as the `.ota` file  generated from the ` .rps `file which is already padded with zero's for alignment.